src/background/libraryManager.ts

```diff
@@ -0,0 +1,231 @@
+      onUpdate({
+        status: "error",
+        message: `ダウンロード先のサーバーがライブラリを返しませんでした`,
+      });
+      return;
+    }
+    const tempFilePath = path.join(
+      this.tempDir,
+      `vv-temp-${engineId}-${libraryId}`
+    );
+    log.log(prefix + `Writing to ${tempFilePath}`);
+
+    const total = Number(downloadRes.headers.get("content-length"));
+    let downloaded = 0;
+    onUpdate({
+      status: "downloading",
+      contentLength: total,
+      downloaded,
+    });
+    const tempFile = fs.createWriteStream(tempFilePath);
+    try {
+      const progressInterval = 1024 * 1024;
+      let lastProgress = 0;
+      downloadRes.body.on("data", (chunk) => {
+        downloaded += chunk.length;
+        tempFile.write(chunk);
+        if (
+          Math.floor(downloaded) - Math.floor(lastProgress) >=
+          progressInterval
+        ) {
+          onUpdate({
+            status: "downloading",
+            contentLength: total,
+            downloaded,
+          });
+          if (total) {
+            log.log(
+              prefix +
+                `Downloaded ${downloaded}/${total} bytes (${(
+                  (downloaded / total) *
+                  100
+                ).toFixed(1)}%)`
+            );
+          } else {
+            log.log(
+              prefix +
+                `Downloaded ${downloaded} bytes (content-length header is not set)`
+            );
+          }
+          lastProgress = downloaded;
+        }
+      });
+      await new Promise((resolve) => {
+        if (!downloadRes.body) throw new Error("res.body is null");
+        downloadRes.body.on("end", resolve);
+      });
+      tempFile.close();
+      log.log(prefix + "Download complete");
+
+      if (this.engineApis[engineId] === undefined) {
+        this.engineApis[engineId] = new DefaultApi(
+          new Configuration({ basePath: engine.host })
+        );
+      }
+      const libraryBuffer = await fs.promises.readFile(tempFilePath);
+
+      onUpdate({
+        status: "installing",
+      });
+      log.log(prefix + "Waiting for lock");
+
+      await this.lock.acquire(`${engineId}`, async () => {
+        log.log(prefix + "Installing library");
+        await this.engineApis[
+          engineId
+        ].installLibraryInstallLibraryLibraryUuidPost(
+          {
+            libraryUuid: libraryId,
+          },
+          {
+            body: libraryBuffer,
+          }
+        );
+        log.log(prefix + "Library installed");
+      });
+      await fs.promises.rm(tempFilePath);
+      onUpdate({
+        status: "done",
+      });
+    } catch (e) {
+      log.error(prefix + "Failed to install library");
+      log.error(e);
+      onUpdate({
+        status: "error",
+        message: `ライブラリのインストールに失敗しました。エラー内容：${e}`,
+      });
+    } finally {
+      tempFile.close();
+      log.log(prefix + "Removing temp file");
+      await fs.promises.rm(tempFilePath);
+    }
+  }
+
+  async uninstallLibrary(
```

Comment:
こっちも正常終了と異常終了でどうなるのか書いてあげると良さそう。今はこうですかね。
```suggestion
  /**
   * 成功時・失敗時ともにステータス情報をonUpdateで通知して正常終了
   */
  async uninstallLibrary(
```